src/Api/Validator.php

```diff
@@ -232,6 +239,21 @@ private function validateRange(Shape $shape, $length, $descriptor)
         }
     }
 
+    private function check_array($arr)
```

Comment:
Minor: Might be a good idea to have this be `checkArray` instead, so it's not easily confusable with something [invoked by `dispatch`](https://github.com/aws/aws-sdk-php/blob/master/src/Api/Validator.php#L69)